{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-08T15:26:40.606067Z",
     "start_time": "2022-03-08T15:26:37.131920Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/tajo5912/jhub_venv/lib/python3.6/site-packages/pandas/compat/__init__.py:120: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.\n",
      "  warnings.warn(msg)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import warnings\n",
    "from pandas.core.common import SettingWithCopyWarning\n",
    "warnings.simplefilter(action=\"ignore\", category=SettingWithCopyWarning)\n",
    "from matplotlib import cm\n",
    "from matplotlib.colors import ListedColormap, LinearSegmentedColormap\n",
    "from matplotlib import pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-08T15:26:40.755816Z",
     "start_time": "2022-03-08T15:26:40.751675Z"
    }
   },
   "outputs": [],
   "source": [
    "workdir = '/scratch/Users/tajo5912/peptide/ChIP-seq/counts/'\n",
    "expt = pd.read_csv(workdir + 'counts_over_TSS_1000bp.txt', sep = '\\t', header=None)\n",
    "shuff= pd.read_csv(workdir + 'shuffle_counts_over_TSS_1000bp.txt', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# normalization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-04T16:35:35.265736Z",
     "start_time": "2022-03-04T16:35:35.261070Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "count_names = ['DMSO_R1','DMSO_R2', 'DMSO_pep_R1','DMSO_pep_R2',\n",
    "        'Nutlin_R1','Nutlin_R2','Nutlin_pep_R1','Nutlin_pep_R2',\n",
    "        'in_DMSO_R1','in_DMSO_R2','in_DMSO_pep_R1','in_DMSO_pep_R2',\n",
    "        'in_Nutlin_R1','in_Nutlin_R2','in_Nutlin_pep_R1','in_Nutlin_pep_R2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-12-01T20:25:09.016727Z",
     "start_time": "2021-12-01T20:25:08.234458Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "82500\n",
      "71996\n",
      "dropped: 10504\n",
      "82500\n",
      "77370\n",
      "dropped: 5130\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>accession</th>\n",
       "      <th>DMSO_R1</th>\n",
       "      <th>DMSO_R2</th>\n",
       "      <th>DMSO_pep_R1</th>\n",
       "      <th>DMSO_pep_R2</th>\n",
       "      <th>Nutlin_R1</th>\n",
       "      <th>Nutlin_R2</th>\n",
       "      <th>...</th>\n",
       "      <th>in_DMSO_R2</th>\n",
       "      <th>in_DMSO_pep_R1</th>\n",
       "      <th>in_DMSO_pep_R2</th>\n",
       "      <th>in_Nutlin_R1</th>\n",
       "      <th>in_Nutlin_R2</th>\n",
       "      <th>in_Nutlin_pep_R1</th>\n",
       "      <th>in_Nutlin_pep_R2</th>\n",
       "      <th>len</th>\n",
       "      <th>sample_counts</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>696</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16740215</td>\n",
       "      <td>16741215</td>\n",
       "      <td>NR_004430_RNU1-1</td>\n",
       "      <td>5033</td>\n",
       "      <td>2243</td>\n",
       "      <td>0</td>\n",
       "      <td>2228</td>\n",
       "      <td>3284</td>\n",
       "      <td>1860</td>\n",
       "      <td>...</td>\n",
       "      <td>38</td>\n",
       "      <td>99</td>\n",
       "      <td>65</td>\n",
       "      <td>49</td>\n",
       "      <td>39</td>\n",
       "      <td>51</td>\n",
       "      <td>60</td>\n",
       "      <td>1000</td>\n",
       "      <td>21637</td>\n",
       "      <td>RNU1-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16740215</td>\n",
       "      <td>16741215</td>\n",
       "      <td>NR_004427_RNU1-2</td>\n",
       "      <td>5033</td>\n",
       "      <td>2243</td>\n",
       "      <td>0</td>\n",
       "      <td>2228</td>\n",
       "      <td>3284</td>\n",
       "      <td>1860</td>\n",
       "      <td>...</td>\n",
       "      <td>38</td>\n",
       "      <td>99</td>\n",
       "      <td>65</td>\n",
       "      <td>49</td>\n",
       "      <td>39</td>\n",
       "      <td>51</td>\n",
       "      <td>60</td>\n",
       "      <td>1000</td>\n",
       "      <td>21637</td>\n",
       "      <td>RNU1-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16740215</td>\n",
       "      <td>16741215</td>\n",
       "      <td>NR_004421_RNU1-4</td>\n",
       "      <td>5033</td>\n",
       "      <td>2243</td>\n",
       "      <td>0</td>\n",
       "      <td>2228</td>\n",
       "      <td>3284</td>\n",
       "      <td>1860</td>\n",
       "      <td>...</td>\n",
       "      <td>38</td>\n",
       "      <td>99</td>\n",
       "      <td>65</td>\n",
       "      <td>49</td>\n",
       "      <td>39</td>\n",
       "      <td>51</td>\n",
       "      <td>60</td>\n",
       "      <td>1000</td>\n",
       "      <td>21637</td>\n",
       "      <td>RNU1-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16740215</td>\n",
       "      <td>16741215</td>\n",
       "      <td>NR_004408_RNU1-3</td>\n",
       "      <td>5033</td>\n",
       "      <td>2243</td>\n",
       "      <td>0</td>\n",
       "      <td>2228</td>\n",
       "      <td>3284</td>\n",
       "      <td>1860</td>\n",
       "      <td>...</td>\n",
       "      <td>38</td>\n",
       "      <td>99</td>\n",
       "      <td>65</td>\n",
       "      <td>49</td>\n",
       "      <td>39</td>\n",
       "      <td>51</td>\n",
       "      <td>60</td>\n",
       "      <td>1000</td>\n",
       "      <td>21637</td>\n",
       "      <td>RNU1-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16740215</td>\n",
       "      <td>16741215</td>\n",
       "      <td>NR_004400_RNVU1-18</td>\n",
       "      <td>5033</td>\n",
       "      <td>2243</td>\n",
       "      <td>0</td>\n",
       "      <td>2228</td>\n",
       "      <td>3284</td>\n",
       "      <td>1860</td>\n",
       "      <td>...</td>\n",
       "      <td>38</td>\n",
       "      <td>99</td>\n",
       "      <td>65</td>\n",
       "      <td>49</td>\n",
       "      <td>39</td>\n",
       "      <td>51</td>\n",
       "      <td>60</td>\n",
       "      <td>1000</td>\n",
       "      <td>21637</td>\n",
       "      <td>RNVU1-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64109</th>\n",
       "      <td>chr6</td>\n",
       "      <td>63319677</td>\n",
       "      <td>63320677</td>\n",
       "      <td>NM_001143940_LGSN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1000</td>\n",
       "      <td>1</td>\n",
       "      <td>LGSN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19493</th>\n",
       "      <td>chr13</td>\n",
       "      <td>86911619</td>\n",
       "      <td>86912619</td>\n",
       "      <td>NR_132371_LINC00430</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1000</td>\n",
       "      <td>1</td>\n",
       "      <td>LINC00430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64125</th>\n",
       "      <td>chr6</td>\n",
       "      <td>68055050</td>\n",
       "      <td>68056050</td>\n",
       "      <td>NR_125869_LOC102723883</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1000</td>\n",
       "      <td>1</td>\n",
       "      <td>LOC102723883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56933</th>\n",
       "      <td>chr4</td>\n",
       "      <td>60922318</td>\n",
       "      <td>60923318</td>\n",
       "      <td>NR_039652_MIR548AG1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1000</td>\n",
       "      <td>1</td>\n",
       "      <td>MIR548AG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13438</th>\n",
       "      <td>chr11</td>\n",
       "      <td>88504275</td>\n",
       "      <td>88505275</td>\n",
       "      <td>NR_049724_GRM5-AS1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1000</td>\n",
       "      <td>1</td>\n",
       "      <td>GRM5-AS1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28159 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr     start      stop               accession  DMSO_R1  DMSO_R2  \\\n",
       "696     chr1  16740215  16741215        NR_004430_RNU1-1     5033     2243   \n",
       "695     chr1  16740215  16741215        NR_004427_RNU1-2     5033     2243   \n",
       "694     chr1  16740215  16741215        NR_004421_RNU1-4     5033     2243   \n",
       "693     chr1  16740215  16741215        NR_004408_RNU1-3     5033     2243   \n",
       "692     chr1  16740215  16741215      NR_004400_RNVU1-18     5033     2243   \n",
       "...      ...       ...       ...                     ...      ...      ...   \n",
       "64109   chr6  63319677  63320677       NM_001143940_LGSN        0        0   \n",
       "19493  chr13  86911619  86912619     NR_132371_LINC00430        0        0   \n",
       "64125   chr6  68055050  68056050  NR_125869_LOC102723883        0        1   \n",
       "56933   chr4  60922318  60923318     NR_039652_MIR548AG1        0        0   \n",
       "13438  chr11  88504275  88505275      NR_049724_GRM5-AS1        0        0   \n",
       "\n",
       "       DMSO_pep_R1  DMSO_pep_R2  Nutlin_R1  Nutlin_R2  ...  in_DMSO_R2  \\\n",
       "696              0         2228       3284       1860  ...          38   \n",
       "695              0         2228       3284       1860  ...          38   \n",
       "694              0         2228       3284       1860  ...          38   \n",
       "693              0         2228       3284       1860  ...          38   \n",
       "692              0         2228       3284       1860  ...          38   \n",
       "...            ...          ...        ...        ...  ...         ...   \n",
       "64109            0            0          0          1  ...           0   \n",
       "19493            0            0          1          0  ...           0   \n",
       "64125            0            0          0          0  ...           0   \n",
       "56933            0            0          1          0  ...           0   \n",
       "13438            0            0          1          0  ...           0   \n",
       "\n",
       "       in_DMSO_pep_R1  in_DMSO_pep_R2  in_Nutlin_R1  in_Nutlin_R2  \\\n",
       "696                99              65            49            39   \n",
       "695                99              65            49            39   \n",
       "694                99              65            49            39   \n",
       "693                99              65            49            39   \n",
       "692                99              65            49            39   \n",
       "...               ...             ...           ...           ...   \n",
       "64109               0               0             0             0   \n",
       "19493               0               1             0             0   \n",
       "64125               0               0             0             0   \n",
       "56933               0               0             0             0   \n",
       "13438               0               0             0             0   \n",
       "\n",
       "       in_Nutlin_pep_R1  in_Nutlin_pep_R2   len  sample_counts          gene  \n",
       "696                  51                60  1000          21637        RNU1-1  \n",
       "695                  51                60  1000          21637        RNU1-2  \n",
       "694                  51                60  1000          21637        RNU1-4  \n",
       "693                  51                60  1000          21637        RNU1-3  \n",
       "692                  51                60  1000          21637      RNVU1-18  \n",
       "...                 ...               ...   ...            ...           ...  \n",
       "64109                 0                 1  1000              1          LGSN  \n",
       "19493                 0                 1  1000              1     LINC00430  \n",
       "64125                 0                 0  1000              1  LOC102723883  \n",
       "56933                 0                 0  1000              1     MIR548AG1  \n",
       "13438                 0                 0  1000              1      GRM5-AS1  \n",
       "\n",
       "[28159 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = ['chr', 'start', 'stop', 'accession', 'score', 'strand', '1', '2', '3', '4', '5', '6']\n",
    "cols.extend(count_names)\n",
    "expt.columns = cols\n",
    "cols = ['chr', 'start', 'stop', 'accession']\n",
    "cols.extend(count_names)\n",
    "expt = expt[cols]\n",
    "\n",
    "scols = ['chr', 'start','stop']\n",
    "scols.extend(count_names)\n",
    "shuff.columns = scols\n",
    "\n",
    "expt['len'] = expt['stop'] - expt['start']\n",
    "shuff['len'] = shuff['stop'] - shuff['start']\n",
    "\n",
    "olenexpt = len(expt)\n",
    "print(olenexpt)\n",
    "\n",
    "expt['sample_counts'] = expt[count_names[0:8]].sum(axis=1)\n",
    "expt = expt[(expt['sample_counts'] > 0)]\n",
    "\n",
    "slenexpt = len(expt)\n",
    "print(slenexpt)\n",
    "print('dropped: ' + str(olenexpt-slenexpt))\n",
    "olenshuff = len(shuff)\n",
    "print(olenshuff)\n",
    "\n",
    "shuff['total_counts'] = shuff[count_names].sum(axis=1)\n",
    "shuff = shuff[(shuff['total_counts'] > 0)]\n",
    "\n",
    "tlenshuff = len(shuff)\n",
    "print(tlenshuff)\n",
    "print('dropped: ' + str(olenshuff-tlenshuff))\n",
    "\n",
    "expt_cut = expt.accession.str.split(\"_\", expand=True,)\n",
    "expt_cut = expt_cut[[2]]\n",
    "expt_cut.columns = ['gene']\n",
    "expt = expt.merge(expt_cut,  left_index=True, right_index=True)\n",
    "\n",
    "expt = expt.sort_values(by='sample_counts', ascending=False)\n",
    "expt= expt.drop_duplicates(subset=['gene'], keep='first')\n",
    "expt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-08T18:15:19.509895Z",
     "start_time": "2022-03-08T18:15:19.503492Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "##pull in millions mapped\n",
    "mm= pd.read_csv('/scratch/Users/tajo5912/peptide/ChIP-seq/qc/chip_millions_mapped.txt', sep='\\t')\n",
    "mm=mm[['sample', 'millionsmapped']]\n",
    "mm = mm.sort_values(by='sample')\n",
    "mm= mm.reset_index(drop=True)\n",
    "mm['labs'] = count_names\n",
    "\n",
    "##pull in p53 targets\n",
    "p53 = pd.read_csv('/Users/tajo5912/temp/p53_target_andrysik2018.txt', sep='\\t', header=None)\n",
    "p53.columns = ['gene', 'pathway']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-12T18:32:07.489575Z",
     "start_time": "2021-10-12T18:32:01.201660Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DPvNP\n",
      "DPvNW\n",
      "DWvDP\n",
      "DWvNP\n",
      "DWvNW\n",
      "NWvNP\n",
      "3135\n",
      "2820\n"
     ]
    }
   ],
   "source": [
    "#filter out all significant genes and p53 genes\n",
    "deseq_dir1 = '/scratch/Users/tajo5912/peptide/RNA-seq/deseq2/expt1/'\n",
    "deseq_dir2 = '/scratch/Users/tajo5912/peptide/RNA-seq/deseq2/expt2_limma_correction/'\n",
    "comparisons = ['DPvNP', 'DPvNW', 'DWvDP', 'DWvNP', 'DWvNW', 'NWvNP']\n",
    "gl=[]\n",
    "for comparison in comparisons:\n",
    "    print(comparison)\n",
    "    sq1 = pd.read_csv(deseq_dir1 + comparison +'/diffexpr-results_DESeq2.txt', sep='\\t')\n",
    "    sq1 = sq1[(sq1['padj'] < 0.05)]\n",
    "    g1 = list(sq1['Gene'])\n",
    "    \n",
    "    sq2 = pd.read_csv(deseq_dir2 + comparison +'/diffexpr-results_DESeq2.txt', sep='\\t')\n",
    "    sq2 = sq2[(sq2['padj'] < 0.05)]\n",
    "    g2 = list(sq2['Gene'])\n",
    "    \n",
    "    gl.extend(g2)\n",
    "    gl.extend(g1)\n",
    "\n",
    "gl.extend(list(p53['gene']))\n",
    "print(len(gl))\n",
    "gs=(set(gl))\n",
    "gl=list(gs)\n",
    "print(len(gl))\n",
    "\n",
    "egl = set(expt['gene'])\n",
    "len(egl.difference(set(gl)))\n",
    "gg = list(egl.difference(set(gl)))\n",
    "len(gg)\n",
    "expt=expt[(expt['gene'].isin(gg))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-12T18:32:13.425975Z",
     "start_time": "2021-10-12T18:32:07.537372Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initially there are 25345 counts in DMSO_R1\n",
      "Now there are 18580 counts remaining.\n",
      "This is 73.30834484119156 percent of the original counts for DMSO_R1\n",
      "Initially there are 25345 counts in DMSO_R2\n",
      "Now there are 17873 counts remaining.\n",
      "This is 70.5188400078911 percent of the original counts for DMSO_R2\n",
      "Initially there are 25345 counts in DMSO_pep_R1\n",
      "Now there are 6 counts remaining.\n",
      "This is 0.02367330834484119 percent of the original counts for DMSO_pep_R1\n",
      "Initially there are 25345 counts in DMSO_pep_R2\n",
      "Now there are 18040 counts remaining.\n",
      "This is 71.17774709015585 percent of the original counts for DMSO_pep_R2\n",
      "Initially there are 25345 counts in Nutlin_R1\n",
      "Now there are 19000 counts remaining.\n",
      "This is 74.96547642533044 percent of the original counts for Nutlin_R1\n",
      "Initially there are 25345 counts in Nutlin_R2\n",
      "Now there are 18381 counts remaining.\n",
      "This is 72.523180114421 percent of the original counts for Nutlin_R2\n",
      "Initially there are 25345 counts in Nutlin_pep_R1\n",
      "Now there are 13255 counts remaining.\n",
      "This is 52.298283685144995 percent of the original counts for Nutlin_pep_R1\n",
      "Initially there are 25345 counts in Nutlin_pep_R2\n",
      "Now there are 17966 counts remaining.\n",
      "This is 70.88577628723614 percent of the original counts for Nutlin_pep_R2\n"
     ]
    }
   ],
   "source": [
    "##getting ratios\n",
    "def no0s(row):\n",
    "    if (row['in_' + i] == 0):\n",
    "        return row['in_' + i] + 0.99999\n",
    "    else:\n",
    "        return row['in_' + i]\n",
    "\n",
    "dd={}\n",
    "lsums_counts=[]\n",
    "lsums_in_norm_counts=[]\n",
    "n_cuts=[]\n",
    "for i in count_names[0:8]:\n",
    "    mini_expt = expt[['chr', 'start', 'stop', 'accession', i, 'in_' + i, 'len']]\n",
    "    mini_expt['in_' + i] = mini_expt.apply(lambda row: no0s(row), axis=1)\n",
    "    mini_expt['in_normalized_' + i] = (mini_expt[i])/(mini_expt['in_'+i])\n",
    "    n_counts = len(mini_expt)\n",
    "    mini_expt = mini_expt[(mini_expt['in_normalized_'+i] > 0)]\n",
    "    a_0 = len(mini_expt)\n",
    "    q=0.25\n",
    "\n",
    "    quant = mini_expt['in_normalized_' + i].quantile(q)\n",
    "    print('Initially there are ' + str(n_counts) + ' counts in ' + i)\n",
    "\n",
    "    mini_expt = mini_expt[(mini_expt['in_normalized_'+i] >= quant)]\n",
    "    n_cut= len(mini_expt)\n",
    "    print('Now there are ' + str(n_cut) + ' counts remaining.')\n",
    "    print('This is ' + str(n_cut/n_counts*100) + ' percent of the original counts for ' + i)\n",
    "\n",
    "    dd[i] = mini_expt\n",
    "    lsums_counts.append(sum(mini_expt[i]))\n",
    "    lsums_in_norm_counts.append(sum(mini_expt['in_normalized_' + i]))\n",
    "    n_cuts.append(n_cut)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-12T18:33:01.983140Z",
     "start_time": "2021-10-12T18:33:01.831116Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Now there are 57337 counts remaining\n",
      "This is 74.10753522036964 percent of the original counts.\n",
      "Now there are 48857 counts remaining\n",
      "This is 63.14721468269355 percent of the original counts.\n",
      "Now there are 0 counts remaining\n",
      "This is 0.0 percent of the original counts.\n",
      "Now there are 50705 counts remaining\n",
      "This is 65.5357373659041 percent of the original counts.\n",
      "Now there are 57933 counts remaining\n",
      "This is 74.87785963551764 percent of the original counts.\n",
      "Now there are 54792 counts remaining\n",
      "This is 70.81814656843738 percent of the original counts.\n",
      "Now there are 14897 counts remaining\n",
      "This is 19.254232906811424 percent of the original counts.\n",
      "Now there are 49166 counts remaining\n",
      "This is 63.54659428719142 percent of the original counts.\n"
     ]
    }
   ],
   "source": [
    "ss={}\n",
    "shuff_lsums_counts=[]\n",
    "shuff_n_cuts=[]\n",
    "for i in count_names[0:8]:\n",
    "    mini_shuff = shuff[['chr', 'start', 'stop', i, 'len']]\n",
    "    n_counts = len(mini_shuff)\n",
    "    lq=0.1\n",
    "    hq = 0.9\n",
    "    low_quant= mini_shuff[i].quantile(lq)\n",
    "    high_quant = mini_shuff[i].quantile(hq)\n",
    "    \n",
    "    mini_shuff = mini_shuff[(mini_shuff[i] > low_quant) & \n",
    "                                (mini_shuff[i] < high_quant)]\n",
    "    n_cut= len(mini_shuff)\n",
    "    print('Now there are ' + str(n_cut) + ' counts remaining')\n",
    "    print('This is ' + str(n_cut/n_counts*100) + ' percent of the original counts.')\n",
    "\n",
    "    ss[i] = mini_shuff\n",
    "    shuff_lsums_counts.append(sum(mini_shuff[i]))\n",
    "    shuff_n_cuts.append(n_cut)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-12T18:33:48.448088Z",
     "start_time": "2021-10-12T18:33:48.406269Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>tss_sum_counts</th>\n",
       "      <th>tss_sum_input_norm_counts</th>\n",
       "      <th>tss_n_cuts</th>\n",
       "      <th>shuff_sum_counts</th>\n",
       "      <th>shuff_n_cuts</th>\n",
       "      <th>tss_av</th>\n",
       "      <th>shuff_av</th>\n",
       "      <th>snr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DMSO_R1</td>\n",
       "      <td>1635572.0</td>\n",
       "      <td>315178.478115</td>\n",
       "      <td>18580.0</td>\n",
       "      <td>333894.0</td>\n",
       "      <td>57337.0</td>\n",
       "      <td>88.028633</td>\n",
       "      <td>5.823360</td>\n",
       "      <td>1.511647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DMSO_R2</td>\n",
       "      <td>1465304.0</td>\n",
       "      <td>203787.990536</td>\n",
       "      <td>17873.0</td>\n",
       "      <td>238366.0</td>\n",
       "      <td>48857.0</td>\n",
       "      <td>81.984222</td>\n",
       "      <td>4.878851</td>\n",
       "      <td>1.680400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DMSO_pep_R1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.592857</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DMSO_pep_R2</td>\n",
       "      <td>1322097.0</td>\n",
       "      <td>104190.509914</td>\n",
       "      <td>18040.0</td>\n",
       "      <td>223818.0</td>\n",
       "      <td>50705.0</td>\n",
       "      <td>73.286973</td>\n",
       "      <td>4.414121</td>\n",
       "      <td>1.660285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Nutlin_R1</td>\n",
       "      <td>933027.0</td>\n",
       "      <td>149941.818860</td>\n",
       "      <td>19000.0</td>\n",
       "      <td>517857.0</td>\n",
       "      <td>57933.0</td>\n",
       "      <td>49.106684</td>\n",
       "      <td>8.938895</td>\n",
       "      <td>0.549360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Nutlin_R2</td>\n",
       "      <td>1224435.0</td>\n",
       "      <td>228473.161728</td>\n",
       "      <td>18381.0</td>\n",
       "      <td>273017.0</td>\n",
       "      <td>54792.0</td>\n",
       "      <td>66.614167</td>\n",
       "      <td>4.982789</td>\n",
       "      <td>1.336885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Nutlin_pep_R1</td>\n",
       "      <td>2557102.0</td>\n",
       "      <td>596055.297735</td>\n",
       "      <td>13255.0</td>\n",
       "      <td>201082.0</td>\n",
       "      <td>14897.0</td>\n",
       "      <td>192.916032</td>\n",
       "      <td>13.498154</td>\n",
       "      <td>1.429203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Nutlin_pep_R2</td>\n",
       "      <td>1688326.0</td>\n",
       "      <td>177197.544578</td>\n",
       "      <td>17966.0</td>\n",
       "      <td>294656.0</td>\n",
       "      <td>49166.0</td>\n",
       "      <td>93.973394</td>\n",
       "      <td>5.993085</td>\n",
       "      <td>1.568030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample  tss_sum_counts  tss_sum_input_norm_counts  tss_n_cuts  \\\n",
       "0        DMSO_R1       1635572.0              315178.478115     18580.0   \n",
       "1        DMSO_R2       1465304.0              203787.990536     17873.0   \n",
       "2    DMSO_pep_R1             6.0                   1.592857         6.0   \n",
       "3    DMSO_pep_R2       1322097.0              104190.509914     18040.0   \n",
       "4      Nutlin_R1        933027.0              149941.818860     19000.0   \n",
       "5      Nutlin_R2       1224435.0              228473.161728     18381.0   \n",
       "6  Nutlin_pep_R1       2557102.0              596055.297735     13255.0   \n",
       "7  Nutlin_pep_R2       1688326.0              177197.544578     17966.0   \n",
       "\n",
       "   shuff_sum_counts  shuff_n_cuts      tss_av   shuff_av       snr  \n",
       "0          333894.0       57337.0   88.028633   5.823360  1.511647  \n",
       "1          238366.0       48857.0   81.984222   4.878851  1.680400  \n",
       "2               0.0           0.0    1.000000        NaN       NaN  \n",
       "3          223818.0       50705.0   73.286973   4.414121  1.660285  \n",
       "4          517857.0       57933.0   49.106684   8.938895  0.549360  \n",
       "5          273017.0       54792.0   66.614167   4.982789  1.336885  \n",
       "6          201082.0       14897.0  192.916032  13.498154  1.429203  \n",
       "7          294656.0       49166.0   93.973394   5.993085  1.568030  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ratios = [lsums_counts, lsums_in_norm_counts, n_cuts, shuff_lsums_counts, shuff_n_cuts]\n",
    "df = pd.DataFrame(ratios, columns = [count_names[0:8]])\n",
    "df = df.transpose().reset_index()\n",
    "df.columns = ['sample' , 'tss_sum_counts', 'tss_sum_input_norm_counts', 'tss_n_cuts',\n",
    "                          'shuff_sum_counts', 'shuff_n_cuts']\n",
    "df['tss_av'] = df['tss_sum_counts']/df['tss_n_cuts']\n",
    "df['shuff_av'] = df['shuff_sum_counts']/df['shuff_n_cuts']\n",
    "df['snr'] = df['tss_av']/df['shuff_av']/10\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-30T16:35:00.727325Z",
     "start_time": "2021-09-30T16:35:00.696690Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "mm.columns = ['file_name', 'millions_mapped', 'sample']\n",
    "df = df.merge(mm, on='sample')\n",
    "df.to_csv('/scratch/Users/tajo5912/peptide/ChIP-seq/annotations/ratio_adj.txt', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-12-01T20:25:42.869643Z",
     "start_time": "2021-12-01T20:25:42.847393Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>file_name_x</th>\n",
       "      <th>snr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DMSO_R1</td>\n",
       "      <td>DMSO_R1_NEW_S3_R1_001</td>\n",
       "      <td>1.511647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DMSO_R2</td>\n",
       "      <td>DMSO_R2_NEW_S4_R1_001</td>\n",
       "      <td>1.680400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DMSO_pep_R1</td>\n",
       "      <td>DMSO_pep_R1_S1_R1_001</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DMSO_pep_R2</td>\n",
       "      <td>DMSO_pep_R2_S2_R1_001</td>\n",
       "      <td>1.660285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Nutlin_R1</td>\n",
       "      <td>Nutlin_R1_NEW_S15_R1_001</td>\n",
       "      <td>0.549360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Nutlin_R2</td>\n",
       "      <td>Nutlin_R2_NEW_S17_R1_001</td>\n",
       "      <td>1.336885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Nutlin_pep_R1</td>\n",
       "      <td>Nutlin_pep_R1_S13_R1_001</td>\n",
       "      <td>1.429203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Nutlin_pep_R2</td>\n",
       "      <td>Nutlin_pep_R2_S14_R1_001</td>\n",
       "      <td>1.568030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample               file_name_x       snr\n",
       "0        DMSO_R1     DMSO_R1_NEW_S3_R1_001  1.511647\n",
       "1        DMSO_R2     DMSO_R2_NEW_S4_R1_001  1.680400\n",
       "2    DMSO_pep_R1     DMSO_pep_R1_S1_R1_001  0.100000\n",
       "3    DMSO_pep_R2     DMSO_pep_R2_S2_R1_001  1.660285\n",
       "4      Nutlin_R1  Nutlin_R1_NEW_S15_R1_001  0.549360\n",
       "5      Nutlin_R2  Nutlin_R2_NEW_S17_R1_001  1.336885\n",
       "6  Nutlin_pep_R1  Nutlin_pep_R1_S13_R1_001  1.429203\n",
       "7  Nutlin_pep_R2  Nutlin_pep_R2_S14_R1_001  1.568030"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ratio_app=pd.read_csv('/scratch/Users/tajo5912/peptide/ChIP-seq/annotations/ratio_adj.txt', sep='\\t')\n",
    "ratio_app = ratio_app[['sample','file_name_x','snr']]\n",
    "ratio_app = ratio_app.fillna(0.1)\n",
    "ratio_app"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-12-01T20:25:53.718653Z",
     "start_time": "2021-12-01T20:25:53.438275Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "convert_dict = {}\n",
    "convert_dict_app_rat = {}\n",
    "for i in count_names[0:8]:\n",
    "    convert_dict[i] = int\n",
    "    expt = expt.astype(convert_dict)\n",
    "    vcut = ratio_app[ratio_app['sample'] == i]\n",
    "    v = float(vcut['snr'])\n",
    "    expt[i+'_applied_ratio'] = expt[i]/v\n",
    "    convert_dict_app_rat[i + '_applied_ratio'] = float\n",
    "    expt = expt.astype(convert_dict_app_rat)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-12-01T20:29:30.127656Z",
     "start_time": "2021-12-01T20:29:29.299987Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "expt.columns\n",
    "# expt.merge(p53, on='gene')\n",
    "exptout=expt[['chr', 'start', 'stop', 'accession', 'DMSO_R1', 'DMSO_R2',\n",
    "       'DMSO_pep_R2', 'Nutlin_R1', 'Nutlin_R2',\n",
    "       'Nutlin_pep_R2', 'in_DMSO_R1', 'in_DMSO_R2', 'in_DMSO_pep_R1',\n",
    "       'in_DMSO_pep_R2', 'in_Nutlin_R1', 'in_Nutlin_R2', 'in_Nutlin_pep_R1',\n",
    "       'in_Nutlin_pep_R2',\n",
    "       'DMSO_R1_applied_ratio', 'DMSO_R2_applied_ratio',\n",
    "       'DMSO_pep_R2_applied_ratio',\n",
    "       'Nutlin_R1_applied_ratio', 'Nutlin_R2_applied_ratio',\n",
    "       'Nutlin_pep_R2_applied_ratio']]\n",
    "exptout.to_csv('/scratch/Users/tajo5912/peptide/ChIP-seq/counts/tss_adjusted_counts.txt', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# bedgraph generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-04T17:31:47.085957Z",
     "start_time": "2022-03-04T17:31:47.056261Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "sample                       DMSO_R1\n",
      "file_name_x    DMSO_R1_NEW_S3_R1_001\n",
      "snr                          1.51165\n",
      "Name: 0, dtype: object\n",
      "1\n",
      "sample                       DMSO_R2\n",
      "file_name_x    DMSO_R2_NEW_S4_R1_001\n",
      "snr                           1.6804\n",
      "Name: 1, dtype: object\n",
      "2\n",
      "sample                   DMSO_pep_R1\n",
      "file_name_x    DMSO_pep_R1_S1_R1_001\n",
      "snr                              0.1\n",
      "Name: 2, dtype: object\n",
      "3\n",
      "sample                   DMSO_pep_R2\n",
      "file_name_x    DMSO_pep_R2_S2_R1_001\n",
      "snr                          1.66028\n",
      "Name: 3, dtype: object\n",
      "4\n",
      "sample                        Nutlin_R1\n",
      "file_name_x    Nutlin_R1_NEW_S15_R1_001\n",
      "snr                             0.54936\n",
      "Name: 4, dtype: object\n",
      "5\n",
      "sample                        Nutlin_R2\n",
      "file_name_x    Nutlin_R2_NEW_S17_R1_001\n",
      "snr                             1.33689\n",
      "Name: 5, dtype: object\n",
      "6\n",
      "sample                    Nutlin_pep_R1\n",
      "file_name_x    Nutlin_pep_R1_S13_R1_001\n",
      "snr                              1.4292\n",
      "Name: 6, dtype: object\n",
      "7\n",
      "sample                    Nutlin_pep_R2\n",
      "file_name_x    Nutlin_pep_R2_S14_R1_001\n",
      "snr                             1.56803\n",
      "Name: 7, dtype: object\n"
     ]
    }
   ],
   "source": [
    "samps=range(8)\n",
    "# ratio_app.loc[samp]\n",
    "for samp in samps:\n",
    "    print(samp)\n",
    "    print(ratio_app.loc[samp])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-30T00:26:17.578354Z",
     "start_time": "2021-09-30T00:19:02.549583Z"
    },
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "for samp in samps:\n",
    "    print('Reading in: ' + ratio_app['file_name'].loc[samp])\n",
    "    print('SNR Value:')\n",
    "    print(float(ratio_app['snr'].loc[samp]))\n",
    "    bg = pd.read_csv('/scratch/Users/tajo5912/peptide/ChIP-seq/mapped/bedgraphs/' + ratio_app['file_name'].loc[samp] + '.bedGraph', \n",
    "                     sep='\\t', header=None)\n",
    "    bg.columns = ['chr', 'start', 'stop', 'level']\n",
    "    bg['level_adj'] = bg['level']/float(ratio_app['snr'].loc[samp])\n",
    "    bg = bg[['chr', 'start', 'stop', 'level_adj']]\n",
    "    outdir = '/scratch/Users/tajo5912/peptide/ChIP-seq/mapped/chipin_adjusted/bedgraphs/'\n",
    "    bg.to_csv(outdir + ratio_app['sample'].loc[samp] + '.bedGraph', sep='\\t', header=None, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# bigwig generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-30T16:41:29.505175Z",
     "start_time": "2021-09-30T16:41:29.497476Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "bedgraph_dir= '/scratch/Users/tajo5912/peptide/ChIP-seq/mapped/chipin_adjusted/bedgraphs/'\n",
    "bigwig_dir = '/scratch/Users/tajo5912/peptide/ChIP-seq/mapped/chipin_adjusted/bigwigs/'\n",
    "chrom_sizes = '/scratch/Users/tajo5912/peptide/ChIP-seq/mapped/ratio_adjusted/hg38.chrom.sizes'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-30T16:49:32.892077Z",
     "start_time": "2021-09-30T16:43:54.801091Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "samps=range(8)\n",
    "for samp in samps:\n",
    "    print('Converting: ' + str(ratio_app['sample'].loc[samp]))\n",
    "    os.system('~/bin/x86_64/./bedGraphToBigWig ' + \n",
    "           bedgraph_dir + str(ratio_app['sample'].loc[samp]) + '.bedGraph ' \n",
    "           + chrom_sizes + ' ' + \n",
    "           bigwig_dir + str(ratio_app['sample'].loc[samp]) + '.bw')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# chip heat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-08T15:26:45.340076Z",
     "start_time": "2022-03-08T15:26:44.832492Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv('/Users/tajo5912/peptide/tss_adjusted_counts.txt', sep='\\t')\n",
    "df_cut = df.accession.str.split(\"_\", expand=True,)\n",
    "df_cut = df_cut[[2]]\n",
    "df_cut.columns = ['gene']\n",
    "df = df.merge(df_cut,  left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dp53 = df.merge(p53, on ='gene')\n",
    "av = ['DMSO','Nutlin']\n",
    "for i in av:\n",
    "    dp53[i + '_ar'] = (dp53[i + '_R1_applied_ratio'] + dp53[i + '_R1_applied_ratio'])/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-26T19:29:18.971759Z",
     "start_time": "2021-10-26T19:29:18.755341Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>accession</th>\n",
       "      <th>DMSO_R1</th>\n",
       "      <th>DMSO_R2</th>\n",
       "      <th>DMSO_pep_R1</th>\n",
       "      <th>DMSO_pep_R2</th>\n",
       "      <th>Nutlin_R1</th>\n",
       "      <th>Nutlin_R2</th>\n",
       "      <th>...</th>\n",
       "      <th>Nutlin_R1_applied_ratio</th>\n",
       "      <th>Nutlin_R2_applied_ratio</th>\n",
       "      <th>Nutlin_pep_R1_applied_ratio</th>\n",
       "      <th>Nutlin_pep_R2_applied_ratio</th>\n",
       "      <th>t</th>\n",
       "      <th>DMSO_ar</th>\n",
       "      <th>Nutlin_ar</th>\n",
       "      <th>Nutlin_pep_ar</th>\n",
       "      <th>np/n</th>\n",
       "      <th>log2_np/n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31975</th>\n",
       "      <td>chr17</td>\n",
       "      <td>74859585</td>\n",
       "      <td>74860585</td>\n",
       "      <td>NM_000835_GRIN2C</td>\n",
       "      <td>38</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>65</td>\n",
       "      <td>60</td>\n",
       "      <td>135</td>\n",
       "      <td>...</td>\n",
       "      <td>109.218064</td>\n",
       "      <td>100.981009</td>\n",
       "      <td>157.430392</td>\n",
       "      <td>146.680822</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>25.138149</td>\n",
       "      <td>109.218064</td>\n",
       "      <td>157.430392</td>\n",
       "      <td>1.343009</td>\n",
       "      <td>0.425469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63506</th>\n",
       "      <td>chr6</td>\n",
       "      <td>36678413</td>\n",
       "      <td>36679413</td>\n",
       "      <td>NM_000389_CDKN1A</td>\n",
       "      <td>221</td>\n",
       "      <td>243</td>\n",
       "      <td>0</td>\n",
       "      <td>205</td>\n",
       "      <td>326</td>\n",
       "      <td>595</td>\n",
       "      <td>...</td>\n",
       "      <td>593.418145</td>\n",
       "      <td>445.064446</td>\n",
       "      <td>960.675236</td>\n",
       "      <td>658.788213</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>146.198180</td>\n",
       "      <td>593.418145</td>\n",
       "      <td>960.675236</td>\n",
       "      <td>1.110159</td>\n",
       "      <td>0.150766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14308</th>\n",
       "      <td>chr11</td>\n",
       "      <td>125569176</td>\n",
       "      <td>125570176</td>\n",
       "      <td>NR_110770_EI24</td>\n",
       "      <td>178</td>\n",
       "      <td>146</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>84</td>\n",
       "      <td>169</td>\n",
       "      <td>...</td>\n",
       "      <td>152.905289</td>\n",
       "      <td>126.413263</td>\n",
       "      <td>281.275634</td>\n",
       "      <td>161.986647</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>117.752380</td>\n",
       "      <td>152.905289</td>\n",
       "      <td>281.275634</td>\n",
       "      <td>1.059392</td>\n",
       "      <td>0.083237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59345</th>\n",
       "      <td>chr5</td>\n",
       "      <td>58459786</td>\n",
       "      <td>58460786</td>\n",
       "      <td>NM_006622_PLK2</td>\n",
       "      <td>312</td>\n",
       "      <td>333</td>\n",
       "      <td>0</td>\n",
       "      <td>357</td>\n",
       "      <td>188</td>\n",
       "      <td>355</td>\n",
       "      <td>...</td>\n",
       "      <td>342.216599</td>\n",
       "      <td>265.542653</td>\n",
       "      <td>417.015616</td>\n",
       "      <td>362.237856</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>206.397430</td>\n",
       "      <td>342.216599</td>\n",
       "      <td>417.015616</td>\n",
       "      <td>1.058505</td>\n",
       "      <td>0.082028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>chr1</td>\n",
       "      <td>18480629</td>\n",
       "      <td>18481629</td>\n",
       "      <td>NM_152375_KLHDC7A</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>14</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>25.484215</td>\n",
       "      <td>16.456164</td>\n",
       "      <td>27.987625</td>\n",
       "      <td>26.147451</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>8.599893</td>\n",
       "      <td>25.484215</td>\n",
       "      <td>27.987625</td>\n",
       "      <td>1.026025</td>\n",
       "      <td>0.037066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64674</th>\n",
       "      <td>chr6</td>\n",
       "      <td>106086035</td>\n",
       "      <td>106087035</td>\n",
       "      <td>NM_001198_PRDM1</td>\n",
       "      <td>29</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>32.765419</td>\n",
       "      <td>14.960149</td>\n",
       "      <td>51.777107</td>\n",
       "      <td>13.392597</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>19.184377</td>\n",
       "      <td>32.765419</td>\n",
       "      <td>51.777107</td>\n",
       "      <td>0.408742</td>\n",
       "      <td>-1.290738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17674</th>\n",
       "      <td>chr12</td>\n",
       "      <td>108561162</td>\n",
       "      <td>108562162</td>\n",
       "      <td>NM_014301_ISCU</td>\n",
       "      <td>71</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>40</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>72.812042</td>\n",
       "      <td>20.196202</td>\n",
       "      <td>16.092885</td>\n",
       "      <td>28.698422</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>46.968646</td>\n",
       "      <td>72.812042</td>\n",
       "      <td>16.092885</td>\n",
       "      <td>0.394144</td>\n",
       "      <td>-1.343206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59647</th>\n",
       "      <td>chr5</td>\n",
       "      <td>73565339</td>\n",
       "      <td>73566339</td>\n",
       "      <td>NM_023039_ANKRA2</td>\n",
       "      <td>114</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>104</td>\n",
       "      <td>66</td>\n",
       "      <td>...</td>\n",
       "      <td>189.311310</td>\n",
       "      <td>49.368493</td>\n",
       "      <td>71.368444</td>\n",
       "      <td>70.151697</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>75.414446</td>\n",
       "      <td>189.311310</td>\n",
       "      <td>71.368444</td>\n",
       "      <td>0.370563</td>\n",
       "      <td>-1.432211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54796</th>\n",
       "      <td>chr3</td>\n",
       "      <td>151316520</td>\n",
       "      <td>151317520</td>\n",
       "      <td>NM_023915_GPR87</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>25.484215</td>\n",
       "      <td>3.740037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.913228</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>1.984591</td>\n",
       "      <td>25.484215</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.075075</td>\n",
       "      <td>-3.735523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45918</th>\n",
       "      <td>chr2</td>\n",
       "      <td>215138326</td>\n",
       "      <td>215139326</td>\n",
       "      <td>NR_103740_ABCA12</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>21.843613</td>\n",
       "      <td>4.488045</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.275485</td>\n",
       "      <td>1.56803</td>\n",
       "      <td>5.292242</td>\n",
       "      <td>21.843613</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.058392</td>\n",
       "      <td>-4.098093</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr      start       stop          accession  DMSO_R1  DMSO_R2  \\\n",
       "31975  chr17   74859585   74860585   NM_000835_GRIN2C       38       50   \n",
       "63506   chr6   36678413   36679413   NM_000389_CDKN1A      221      243   \n",
       "14308  chr11  125569176  125570176     NR_110770_EI24      178      146   \n",
       "59345   chr5   58459786   58460786     NM_006622_PLK2      312      333   \n",
       "738     chr1   18480629   18481629  NM_152375_KLHDC7A       13       12   \n",
       "...      ...        ...        ...                ...      ...      ...   \n",
       "64674   chr6  106086035  106087035    NM_001198_PRDM1       29       20   \n",
       "17674  chr12  108561162  108562162     NM_014301_ISCU       71       27   \n",
       "59647   chr5   73565339   73566339   NM_023039_ANKRA2      114       88   \n",
       "54796   chr3  151316520  151317520    NM_023915_GPR87        3        2   \n",
       "45918   chr2  215138326  215139326   NR_103740_ABCA12        8        0   \n",
       "\n",
       "       DMSO_pep_R1  DMSO_pep_R2  Nutlin_R1  Nutlin_R2  ...  \\\n",
       "31975            0           65         60        135  ...   \n",
       "63506            0          205        326        595  ...   \n",
       "14308            0          133         84        169  ...   \n",
       "59345            0          357        188        355  ...   \n",
       "738              0           20         14         22  ...   \n",
       "...            ...          ...        ...        ...  ...   \n",
       "64674            0           18         18         20  ...   \n",
       "17674            0           33         40         27  ...   \n",
       "59647            0           61        104         66  ...   \n",
       "54796            0            4         14          5  ...   \n",
       "45918            0            0         12          6  ...   \n",
       "\n",
       "       Nutlin_R1_applied_ratio  Nutlin_R2_applied_ratio  \\\n",
       "31975               109.218064               100.981009   \n",
       "63506               593.418145               445.064446   \n",
       "14308               152.905289               126.413263   \n",
       "59345               342.216599               265.542653   \n",
       "738                  25.484215                16.456164   \n",
       "...                        ...                      ...   \n",
       "64674                32.765419                14.960149   \n",
       "17674                72.812042                20.196202   \n",
       "59647               189.311310                49.368493   \n",
       "54796                25.484215                 3.740037   \n",
       "45918                21.843613                 4.488045   \n",
       "\n",
       "       Nutlin_pep_R1_applied_ratio  Nutlin_pep_R2_applied_ratio        t  \\\n",
       "31975                   157.430392                   146.680822  1.56803   \n",
       "63506                   960.675236                   658.788213  1.56803   \n",
       "14308                   281.275634                   161.986647  1.56803   \n",
       "59345                   417.015616                   362.237856  1.56803   \n",
       "738                      27.987625                    26.147451  1.56803   \n",
       "...                            ...                          ...      ...   \n",
       "64674                    51.777107                    13.392597  1.56803   \n",
       "17674                    16.092885                    28.698422  1.56803   \n",
       "59647                    71.368444                    70.151697  1.56803   \n",
       "54796                     0.000000                     1.913228  1.56803   \n",
       "45918                     0.000000                     1.275485  1.56803   \n",
       "\n",
       "          DMSO_ar   Nutlin_ar  Nutlin_pep_ar      np/n  log2_np/n  \n",
       "31975   25.138149  109.218064     157.430392  1.343009   0.425469  \n",
       "63506  146.198180  593.418145     960.675236  1.110159   0.150766  \n",
       "14308  117.752380  152.905289     281.275634  1.059392   0.083237  \n",
       "59345  206.397430  342.216599     417.015616  1.058505   0.082028  \n",
       "738      8.599893   25.484215      27.987625  1.026025   0.037066  \n",
       "...           ...         ...            ...       ...        ...  \n",
       "64674   19.184377   32.765419      51.777107  0.408742  -1.290738  \n",
       "17674   46.968646   72.812042      16.092885  0.394144  -1.343206  \n",
       "59647   75.414446  189.311310      71.368444  0.370563  -1.432211  \n",
       "54796    1.984591   25.484215       1.000000  0.075075  -3.735523  \n",
       "45918    5.292242   21.843613       1.000000  0.058392  -4.098093  \n",
       "\n",
       "[97 rows x 37 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dp53['Nutlin_ar']=dp53['Nutlin_ar'].replace(0,1)\n",
    "dp53['Nutlin_pep_R2_applied_ratio']=dp53['Nutlin_pep_R2_applied_ratio'].replace(0,1)\n",
    "\n",
    "dp53['np/n'] = dp53['Nutlin_pep_R2_applied_ratio']/dp53['Nutlin_ar']\n",
    "dp53['log2_np/n'] = np.log2(dp53['np/n'])\n",
    "dp53.describe()\n",
    "dp53=dp53.sort_values(by='log2_np/n', ascending=False)\n",
    "dp53=dp53[(dp53['gene'].isin(p53_genes))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-26T19:29:21.951681Z",
     "start_time": "2021-10-26T19:29:21.856741Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "viridis = cm.get_cmap('viridis', 256)\n",
    "newcolors = viridis(np.linspace(0, 1, 256))\n",
    "pink = np.array([248/256, 24/256, 148/256, 1])\n",
    "newcolors[:25, :] = pink\n",
    "newcmp = ListedColormap(newcolors)\n",
    "\n",
    "\n",
    "def plot_examples(cms):\n",
    "    \"\"\"\n",
    "    helper function to plot two colormaps\n",
    "    \"\"\"\n",
    "    np.random.seed(19680801)\n",
    "    data = np.random.randn(30, 30)\n",
    "\n",
    "    fig, axs = plt.subplots(1, 2, figsize=(6, 3), constrained_layout=True)\n",
    "    for [ax, cmap] in zip(axs, cms):\n",
    "        psm = ax.pcolormesh(data, cmap=cmap, rasterized=True, vmin=-4, vmax=4)\n",
    "        fig.colorbar(psm, ax=ax)\n",
    "    plt.show()\n",
    "\n",
    "#plot_examples([viridis, newcmp])\n",
    "\n",
    "\n",
    "top = cm.get_cmap('Blues_r', 128)\n",
    "bottom = cm.get_cmap('YlOrBr', 128)\n",
    "\n",
    "newcolors = np.vstack((top(np.linspace(0, 1, 128)),\n",
    "                       bottom(np.linspace(0, 1, 128))))\n",
    "newcmp = ListedColormap(newcolors, name='OrangeBlue')\n",
    "#plot_examples([viridis, newcmp])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-10-26T19:29:28.874055Z",
     "start_time": "2021-10-26T19:29:23.578044Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "plt = sns.clustermap(dp53[['log2_np/n']], cmap=newcmp, \n",
    "                     yticklabels=dp53['gene'], method='ward', figsize=(8, 16), \n",
    "                     row_cluster=False, col_cluster=False, vmin=-4.7, vmax=4.7)\n",
    "\n",
    "plt.savefig(workdir+'R12Nut_R2Nutpep_logFC_chip_p53.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "applied_ratio_names = []\n",
    "for i in count_names[0:8]:\n",
    "    applied_ratio_names.append(i+'_applied_ratio')\n",
    "dp53['sum_applied_ratio'] = dp53[applied_ratio_names].sum(axis=1)\n",
    "\n",
    "plt = sns.clustermap(dp53[['sum_applied_ratio']], cmap='Greys', \n",
    "                     yticklabels=dp53['gene'], method='ward', figsize=(8, 16), \n",
    "                     row_cluster=False, col_cluster=False, vmin=0, vmax=3100)\n",
    "\n",
    "plt.savefig(workdir+'read_density_chip_p53.pdf')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
